client/src/templates/Challenges/utils/worker-executor.ts

```diff
@@ -26,12 +40,13 @@ class WorkerExecutor {
         if (e.data?.type === 'contentLoaded') {
           resolve(newWorker);
         }
+        // do something else if not contentLoaded?
```

Comment:
i'm in search of some advice. getworker is supposed to return a worker by calling createworker. but if createworker doesn’t get ‘contentloaded’ or has an error, then it won’t return a worker. Right now the failure to get ‘contentloaded’ would result in the call to getWorker inside execute never resolving and the whole thing getting stuck. Typescript is forcing me to address this more directly and so I’ve gone down this path but I wanted some feedback if it seems reasonable?

Reply:
You can see it get stuck by changing the test. If the mockworker doesn't call 'contentloaded', the whole thing will hang till jest shuts it down.

Reply:
i know @ojeytonwilliams  had worked on this code before. is it cool to tag you for advice here?

Reply:
I'm afraid I've never understood this code particularly well, so I can't give particularly useful advice.

All I can say is that, despite being confusing, this code works well. So, if you need to use `@ts-expect-error` to keep TS from complaining, that's totally fine.